qiskit/circuit/classicalfunction/boolean_expression.py

```diff
@@ -0,0 +1,173 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A quantum oracle constructed from a logical expression or a string in the DIMACS format."""
+
+from os.path import basename
+
+from qiskit.circuit import Gate
+from qiskit.exceptions import QiskitError
+from .classicalfunction import HAS_TWEEDLEDUM
+from .utils import tweedledum2qiskit
+
+
+class BooleanExpression(Gate):
+    r"""The Logical Expression Quantum Oracle.
```

Comment:
I think this docstring should be in the `PhaseOracle` class, since that's the replacement of the previous `LogicalExpressionOracle` where this docstring comes from 🙂

Reply:
Yeap! Moved.